pkg/rc/replication_controller.go

```diff
@@ -226,19 +229,19 @@ func (rc *replicationController) addPods(current types.PodLocations) error {
 
 	rc.logger.NoFields().Infof("Need to schedule %d nodes out of %s", toSchedule, possible)
 
-	ctx, cancelFunc := transaction.New(context.Background())
+	txn, cancelFunc := rc.newAuditingTransaction(context.Background(), currentNodes)
 	for i := 0; i < toSchedule; i++ {
 		// create a new context for every 5 nodes. This is done to make sure
```

Comment:
Unrelated to the PR here, but would you mind adding a link to this issue in this comment? https://github.com/hashicorp/consul/issues/2921
We'll be able to simplify this code soon.